src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -300,6 +301,25 @@ async def create_dataset_records(
         raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(err))
 
 
+@router.post("/datasets/{dataset_id}/records/bulk-delete", response_model=BulkDeleteResponse)
+async def bulk_delete_dataset_records(
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    dataset_id: UUID,
+    bulk_delete: BulkDelete,
+    current_user: User = Security(auth.get_current_user),
+):
+    dataset = await _get_dataset(db, dataset_id)
+
+    await authorize(current_user, DatasetPolicyV1.bulk_delete_records(dataset))
+
+    deleted_records_ids = await datasets.bulk_delete_records(db, dataset=dataset, records_ids=bulk_delete.ids)
+
+    return BulkDeleteResponse(
+        deleted=deleted_records_ids, not_deleted=list(set(bulk_delete.ids) - set(deleted_records_ids))
+    )
```

Comment:
I though that it might be useful to return which records were deleted and which not (the user can provide IDs that don't exist), but maybe we can return an empty response and a 204.

Reply:
For me it depends. Why some of the records have not been deleted, Did any error occur? In that case, I would say to return a 400 or 422 error with failed record ids. If everything goes well, I think the 204 is enough as a response.